contracts/pol/CarbonPOL.sol

```diff
@@ -128,18 +137,47 @@ contract CarbonPOL is ICarbonPOL, Upgradeable, ReentrancyGuardUpgradeable, Utils
     }
 
     /**
-     * @notice enable trading for a token and set the initial price
+     * @notice sets the eth sale amount
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setEthSaleAmount(uint128 newEthSaleAmount) external onlyAdmin greaterThanZero(newEthSaleAmount) {
+        _setEthSaleAmount(newEthSaleAmount);
+    }
+
+    /**
+     * @notice enable trading for TKN->ETH and set the initial price
      *
      * requirements:
      *
      * - the caller must be the admin of the contract
+     * - can only enable trading for non-native tokens
      */
     function enableTrading(Token token, Price memory price) external onlyAdmin validPrice(price) {
+        if (token == NATIVE_TOKEN) {
+            revert InvalidToken();
+        }
         _tradingStartTimes[token] = uint32(block.timestamp);
```

Comment:
There weren't any ETH trades enabled already, right? If there were, then I'd consider explicitly "resetting" everything re: ETH during the upgrade

Reply:
I don't think so, but that's a good note - this is why this change was added - enabling ETH from this function messes up the new logic. Worth checking imo @yudilevi

Reply:
not following you but no ETH -> BNT trading was ever enabled

Reply:
added this check since there is explicit eth sale amount logic embedded in the new function enableTradingETH and also since the new function is used to enable trading for eth, so it doesn't make sense to be able to enable eth trading from both imo